src/AzureStorage/Queues/AzureStorageQueue.cs

```diff
@@ -32,7 +36,7 @@ public class AzureStorageQueue<T> : QueueBase<T> where T : class {
 
             _queueName = queueName;
             _queueReference = client.GetQueueReference(queueName);
-            _deadletterQueueReference = client.GetQueueReference(String.Concat(queueName, "-deadletter"));
+            _deadletterQueueReference = client.GetQueueReference($"{queueName}-deadletter");
```

Comment:
Does this do anything network related? If so, I wasn't sure if this should be moved to the ensure method.